sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -102,6 +105,28 @@ case class ScalaUDF(
     }
   }
 
+  /**
+   * Create the converter which converts the scala data type to the catalyst data type for
+   * the return data type of udf function. We'd use `ExpressionEncoder` to create the
+   * converter for typed ScalaUDF only, since its the only case where we know the  type tag
+   * of the return data type of udf function.
+   * @param dataType return type of function
+   * @return the catalyst converter
+   */
+  private def catalystConverter(dataType: DataType): Any => Any = {
+    if (returnEncoder.isDefined) {
```

Comment:
nit format: how about using `.map` instead of `isDefined`?
```
    returnEncoder.map { enc =>
      val toRow = enc.createSerializer()
      if (enc.isSerializedAsStruct) {
```